### PR TITLE
[UT-287] Separate steem account creation

### PR DESCRIPTION
**Pull request by icaroharry**
### Verify that
- [x] the branch name follows the git flow naming convention (feature/i18n)
- [x] the UT reference is appended to the branch name if it concerns a Jira Task (feature/i18n/UT-283)
- [x] the design is responsive
- [x] the code doesn't break Server Side Rendering
- [x] new functions are prefixed with JSDoc information
- [x] the pull request contains unit tests for new code that passes and has at least 50% coverage
- [x] the code doesn't break the api endpoints contract
- [x] JS Standard code style has been verified with ESLint

This PR contains the base for Steem account creation separation from the signup.
Changes:
- Created a base layout for all the pages in the auth app (including the Steem creation ones):
image
- Implemented `hasClaimedBlockchainAccount` endpoint + tests + store actions
- Implemented route auth validation as @gregory-latinier did on the client app
- Created `steem/connect` page
- Created the base of `steem/create` page with the proper validations (auth & hasClaimed)

### **Test plan**

Route validation

* As a guest, go to `localhost:8081/en/steem/connect`

**Expected result:** You should be redirected to `localhost:8081/en/login`
___
Steem create page

* As a logged user, go to `localhost:8081/en/steem/connect`
* Click on "I don't have a steem account"
* You should be sent to `localhost:8081/en/steem/create`

**Expected result:** The page should mount and show the wip text 
___
Steem create page - user has claimed blockchain account
* Insert a new register in the db using your account data
```
db.utopianBlockchainAccounts.insertOne({ "blockchain": "steem", "address": "yourSteemAddress",  "provider": "github", "username": "yourGithubUsername" })
```
* Go to `localhost:8081/en/steem/create`
*  A modal should appear telling you can't create the account.
* Click on 'go back'

**Expected result:** you should be redirected to the main app

### CODE REVIEW

File: packages/api/modules/users/handlers.js [hunk 1]
```diff
@@ -11,6 +12,30 @@ const getUserByUsername = async (req, h) => {
...
+    '$or': user.authProviders.map((authProvider) => ({
+      '$and': [{
+        provider: authProvider.type,
+        username: authProvider.username
+      }]
+    }))
+  })
+
+  // Parse result to boolean
+  return !!parseInt(hasClaimed)
```
Comment (gregory-latinier) [inline packages/api/modules/users/handlers.js]: How about  `hasClaimed > 0` ?
No need for the comment then :)

File: packages/api/modules/users/handlers.js [hunk 2]
```diff
@@ -11,6 +12,30 @@ const getUserByUsername = async (req, h) => {
...
+ *  @param {string} req.params.blockchain - which blockchain. Currently only accepts 'steem'
+ *  @returns {boolean}
+ *  @author Icaro Harry
+ */
+const hasClaimedBlockchainAccount = async (req, h) => {
+  const user = await User.findOne({ _id: req.auth.credentials.uid })
+
+  const hasClaimed = await UtopianBlockchainAccounts.count({
+    blockchain: req.params.blockchain,
+    '$or': user.authProviders.map((authProvider) => ({
```
Comment (gregory-latinier) [inline packages/api/modules/users/handlers.js]: No need for the quotes

File: packages/api/modules/users/handlers.js [hunk 3]
```diff
@@ -11,6 +12,30 @@ const getUserByUsername = async (req, h) => {
...
+      '$and': [{
+        provider: authProvider.type,
+        username: authProvider.username
+      }]
```
Comment (gregory-latinier) [inline packages/api/modules/users/handlers.js]: You can add a test to userId too

Comment (icaroharry) [reply packages/api/modules/users/handlers.js]: you're right. I forgot about that one

**Comment by icaroharry** [comment]
@gregory-latinier please check if you're running the correct code for this PR, I don't have these issues locally

**Comment by nothingismagick** [comment]
For the sake of DB consistency in the future and potentially strange regression behaviour, wouldn't it make sense to prune the DB?

screen shot 2018-11-27 at 19 03 21

**Comment by icaroharry** [comment]
You're right. Just did it :)

**Comment by nothingismagick** [comment]
The language is strange here. We shouldn't be sorry, because we are happy. They already have linked their account. (Unless something fishy is going on, which they ABSOLUTELY should report - and this would be the place to do that IMHO.)

screen shot 2018-11-27 at 19 07 32

## CONGRATULATIONS!
**You already linked your account with Utopian.io!**
*You didn't? [Send us a message](#nowhere).*

`RETURN TO THE APP`

**Comment by icaroharry** [comment]
Just to clarify. This is the steem account creation page. In this case, the user is trying to create a steem account using the Utopian claimed accounts. We limit the account creation to 1 account per user.

The endpoint verifies if the user already claimed any accounts before and if he did, this message is displayed

**Pull request merged**